homeassistant/components/omnilogic/sensor.py

```diff
@@ -0,0 +1,688 @@
+"""Definition and setup of the Omnilogic Sensors for Home Assistant."""
+
+import logging
+
+from homeassistant.const import PERCENTAGE, TEMP_CELSIUS, TEMP_FAHRENHEIT
+
+from .const import COORDINATOR, DOMAIN
+from .omnilogic_common import OmniLogicEntity, OmniLogicUpdateCoordinator
+
+TEMP_UNITS = [TEMP_CELSIUS, TEMP_FAHRENHEIT]
+PERCENT_UNITS = [PERCENTAGE, PERCENTAGE]
+SALT_UNITS = ["g/L", "ppm"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities, discovery_info=None):
+    """Set up the sensor platform."""
+
+    coordinator = hass.data[DOMAIN][entry.entry_id][COORDINATOR]
+    sensors = []
+
+    for backyard in coordinator.data:
+        sensors.append(
+            OmniLogicTemperatureSensor(
+                coordinator,
+                "air_temperature",
+                "Air Temperature",
+                backyard,
+                {},
+                "temperature",
+                "mdi:thermometer",
+                TEMP_UNITS,
+                backyard.get("airTemp"),
+            )
+        )
+
+        sensors.append(
+            OmniLogicAlarmSensor(
+                coordinator,
+                "alarm",
+                "Alarm",
+                backyard,
+                {},
+                "none",
+                "mdi:alert-circle",
+                "",
+                backyard,
+            )
+        )
+
+        for bow in backyard["BOWS"]:
+            sensors.append(
+                OmniLogicTemperatureSensor(
+                    coordinator,
+                    "water_temperature",
+                    "Water Temperature",
+                    backyard,
+                    bow,
+                    "temperature",
+                    "mdi:thermometer",
+                    TEMP_UNITS,
+                    bow.get("waterTemp"),
+                )
+            )
+
+            if "Filter" in bow.keys():
```

Comment:
The keys is the default iterator of a dict.
```suggestion
            if "Filter" in bow:
```

Reply:
Addressed in next commit.